hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.test.tsx

```diff
@@ -0,0 +1,170 @@
+import React from "react";
+import ProductOverview, { EnhancedAddToCartForm } from "./ProductOverview";
+import { mockCategories, mockHardware } from "testing/mockData";
+import { render, fireEvent, waitFor, makeStoreWithEntities } from "testing/utils";
+import { makeStore } from "slices/store";
+import { addToCart, cartSelectors } from "slices/hardware/cartSlice";
+
+describe("<ProductOverview />", () => {
+    test("all 3 parts of the product overview is there", () => {
+        const store = makeStore({
+            ui: {
+                inventory: {
+                    hardwareItemBeingViewed: mockHardware[0],
+                    isProductOverviewVisible: true,
+                },
+            },
+        });
+
+        const { getByText } = render(<ProductOverview showAddToCartButton />, {
+            store,
+        });
+
+        // Check if the main section, detailInfoSection, and add to cart section works
+        expect(getByText("Category")).toBeInTheDocument();
+        expect(getByText("Datasheet")).toBeInTheDocument();
+        expect(getByText("Add to cart")).toBeInTheDocument();
+    });
+
+    test("minimum constraint between hardware and categories is used", () => {
+        const store = makeStoreWithEntities({
+            hardware: [mockHardware[0]],
+            categories: mockCategories,
+            ui: {
+                inventory: {
+                    hardwareItemBeingViewed: mockHardware[0],
+                    isProductOverviewVisible: true,
+                },
+            },
+        });
+
+        const minConstraint: number = mockCategories
+            .map((category) => category.max_per_team)
+            .concat([mockHardware[0].max_per_team])
+            .reduce((prev, curr) => Math.min(prev, curr));
+
+        const { getByText, getByRole, queryByText } = render(
+            <ProductOverview showAddToCartButton={true} />,
+            {
+                store,
+            }
+        );
+
+        fireEvent.mouseDown(getByRole("button", { name: "Qty 1" }));
```

Comment:
will remove

Reply:
hmm actually I think this is the dropdown role

Reply:
Seems you're right because the tests break if it's removed, but I don't see anything with that role unless mui is doing it under the hood? @lisa-sa-li do you know anything about this?